lib/src/script.dart

```diff
@@ -480,7 +541,7 @@ class Script {
   /// If [silenceStartMessage] is `false` (the default), this prints a message
   /// in debug mode indicating that the script has started running.
   Script._(this.name, StreamConsumer<List<int>> stdin, Stream<List<int>> stdout,
-      Stream<List<int>> stderr, Future<int> exitCode,
+      Stream<List<int>> stderr, Future<ScriptComponents> components,
```

Comment:
This new signature means you're passing in a bunch of these values twice: once directly and once through `components`. The whole idea of this constructor is to avoid using `ScriptComponents` in the first place—so can it just take `_signalHandler` as a direct argument?

Reply:
So unfortunately given the way the code is structured I would have to either a) convert to a `factory` constructor which then requires modifying how `BufferedScript` works because it calls `super.fromComponents`, or b) create another holding class for both the exit-code and signal-handler

I would prefer (a) except that it makes inheritance for BufferedScript harder, so I will go with (b)

Reply:
or... I could use a completer 🤔 yeah that may be nicer